backend/climateconnect_api/views/user_views.py

```diff
@@ -45,18 +52,38 @@ def post(self, request):
             raise ValidationError("Email already in use.")
 
         user = User.objects.create(
-            username=request.data['email'], password=request.data['password'],
+            username=request.data['email'],
             email=request.data['email'], first_name=request.data['first_name'],
             last_name=request.data['last_name'], is_active=True
         )
 
+        user.set_password(request.data['password'])
+        user.save()
+
+        url_slug = (user.first_name + user.last_name).lower()
```

Comment:
This can stay like this for now, but maybe add a TODO comment to add some numbers incase the user slug already exists